include/onnxruntime/core/session/environment.h

```diff
@@ -99,5 +124,6 @@ class Environment {
   std::unique_ptr<onnxruntime::concurrency::ThreadPool> inter_op_thread_pool_;
   bool create_global_thread_pools_{false};
   std::vector<AllocatorPtr> shared_allocators_;
+  std::unordered_map<std::string, std::unique_ptr<ProviderLibrary2>> external_ep_libs_;
```

Comment:
did we observe any issue when the dll get unloaded during process shutdown?

Reply:
ProviderLibrary2 is currently used for the EP shared library built with provider_bridges approach, like cuda EP. here you are reusing it for your custom EP which is implemented with a different approach. I am not sure whether it is fully compatible. Maybe decouple the implementation is not a bad idea.